include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -480,12 +496,28 @@ MultigridPreconditionerBase<dim, Number>::do_initialize_dof_handler_and_constrai
       // setup dof_handler: create dof_handler...
       auto dof_handler = new dealii::DoFHandler<dim>(*tria);
       // ... create FE and distribute it
-      if(level.is_dg)
-        dof_handler->distribute_dofs(
-          dealii::FESystem<dim>(dealii::FE_DGQ<dim>(level.degree), n_components));
+      // TODO: following if-else is not used becuade of the above Assert
```

Comment:
As long as we cannot remove the Assert above we should keep the old code. Simplex code only makes sense in code paths where simplex is allowed.